plugin/src/middleware/request.ts

```diff
@@ -21,15 +21,16 @@ type AugmentedGeo = NextRequest['geo'] & {
 /**
  * Supercharge your Next middleware with Netlify Edge Functions
  */
-export class MiddlewareRequest {
+export class MiddlewareRequest extends Request {
   context: Context
   originalRequest: Request
 
-  constructor(public request: NextRequest) {
+  constructor(private nextRequest: NextRequest) {
+    super(nextRequest)
     if (!('Deno' in globalThis)) {
       throw new Error('NetlifyMiddleware only works in a Netlify Edge Function environment')
     }
-    const geo = request.geo as AugmentedGeo
+    const geo = nextRequest.geo as AugmentedGeo
```

Comment:
[dust/sand] Consider extending `NextRequest` with `geo` typed as `AugmentedGeo` and have that be called something like `NetlifyRequest`, or `EdgeRequest`